checkstyle.xml

```diff
@@ -0,0 +1,213 @@
+<?xml version="1.0"?>
```

Comment:
I'm not keen on adding yet another top-level file in our repo. Can it go somewhere else?

Reply:
I've looked for a place, but didn't found one that would make sense. It could be used on eclipse as well and probably on netbeans.

Reply:
@kevinrushforth can you guide me what should I do about this file?

Reply:
Since we already have a few other build and IDE-related files in the top repo, and since it could be shared by other IDEs, I won't object to putting it in the top repo.